core-bundle/src/Routing/LegacyRouteProvider.php

```diff
@@ -14,33 +14,31 @@
 
 use Symfony\Cmf\Component\Routing\RouteProviderInterface;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
 use Symfony\Component\Routing\Route;
 use Symfony\Component\Routing\RouteCollection;
 
+@trigger_error('The following routes are deprecated: contao_frontend, contao_index, contao_root, contao_catch_all', E_USER_DEPRECATED);
```

Comment:
Did you make sure that the deprecation message is only triggered if one of the deprecated routes is used?

Reply:
It was only triggered when the class is loaded, which shouldn't happen usually. But I improved it in f7ae7a9e4fc1ac5cfef4f906407884431ad7e87c